apps/emqx_rule_engine/src/emqx_rule_funcs.erl

```diff
@@ -776,14 +774,7 @@ sha256(S) when is_binary(S) ->
     hash(sha256, S).
 
 hash(Type, Data) ->
-    hexstring(crypto:hash(Type, Data)).
-
-hexstring(<<X:128/big-unsigned-integer>>) ->
-    iolist_to_binary(io_lib:format("~32.16.0b", [X]));
```

Comment:
warn: old implementation uses `a-f`. new implementation uses `A-F`
if this ID is persisted, upgrade may fail.